firedrake/functionspace.py

```diff
@@ -225,7 +225,8 @@ def MixedFunctionSpace(spaces, name=None, mesh=None):
 
         def rec(eles):
             for ele in eles:
-                if ele.num_sub_elements() > 0:
+                if ele.num_sub_elements() > 0 \
```

Comment:
There was some related change in UFL I had to make when merging our UFL upstream. IIRC `TensorProductElement` now also stores its subelements this way.